lib/mayaUsd/fileio/jobs/writeJob.cpp

```diff
@@ -163,10 +163,74 @@ bool UsdMaya_WriteJob::Write(const std::string& fileName, bool append)
     return true;
 }
 
+static MStringArray GetExportDefaultPrimCandidates(const UsdMayaJobExportArgs& exportArgs)
+{
+    MStringArray roots;
+
+    // If the user provided a root prim, used it as the default prim.
+    if (!exportArgs.rootPrim.IsEmpty()) {
+        roots.append(exportArgs.rootPrim.GetName().c_str());
```

Comment:
on the check a little bit below, we check if the string is not empty. We don't do that here, would that be necessary?

Reply:
The IsEmpty(0 is equivalent. IsEmpty is on TfToken, empty is on std::string. A non-empty TfToken is a non-empty std::string.

Reply:
Ah, right.. I got confused with the array